File: backend/src/main/java/com/hyun/moondab/controller/question/QuestionController.java
package com.hyun.moondab.controller.question;

import com.hyun.moondab.application.question.QuestionService;
import com.hyun.moondab.controller.question.dto.QuestionDto;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/questions")
public class QuestionController {

    private final QuestionService questionService;

    @GetMapping("/{id}")
    public QuestionDto getQuestion(@PathVariable Long id) {
        return questionService.getQuestion(id);
    }

}

File: backend/src/main/kotlin/com/hyun/moondab/controller/QuestionController.kt
package com.hyun.moondab.controller

import com.hyun.moondab.domain.QuestionResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/questions")
class QuestionController {

    @GetMapping("/{id}")
    fun getQuestion(@PathVariable id: Long) : QuestionResponse {
        return QuestionResponse(id);
    }

}